go-controller/pkg/allocator/mac/reservation_test.go

```diff
@@ -0,0 +1,75 @@
+package mac_test
+
+import (
+	"net"
+
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/allocator/mac"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("ReservationManager", func() {
+	const owner1 = "namespace1/pod1"
+	const owner2 = "namespace2/pod2"
+
+	var testMgr *mac.ReservationManager
+	var mac1 net.HardwareAddr
+
+	BeforeEach(func() {
+		var err error
+		mac1, err = net.ParseMAC("aa:bb:cc:dd:ee:f1")
+		Expect(err).NotTo(HaveOccurred())
+
+		testMgr = mac.NewManager()
+	})
+
+	Context("reserve", func() {
+		DescribeTable("should not reserve MAC, given",
+			func(owner string, mac net.HardwareAddr) {
+				Expect(testMgr.Reserve(owner, mac)).To(Succeed())
+
+				Expect(testMgr.Reserve(owner1, mac1)).To(Succeed(), "mac should not be reserved")
```

Comment:
we can't really know from the test if it got reserved or not. 
```suggestion
				Expect(testMgr.Reserve(owner1, mac1)).To(Succeed()
```

please consider adding a check that the test indeed doesn't save these values.
Alternatively you can change the Context name to "should return nil, given"

Reply:
Assertion message was fixed.

Reply:
Following offline discussion, input validation checks were removed and the related tests. Input is validated at the caller.